middleware/fstype_test.go

```diff
@@ -0,0 +1,24 @@
+package middleware
+
+import (
+	"runtime"
+	"testing"
+)
+
+func TestCaseSensitiveFS(t *testing.T) {
+	caseSensitive := false
+	switch runtime.GOOS {
+	case "linux":
+		caseSensitive = true
+	case "darwin":
+		caseSensitive = false
```

Comment:
Some Mac users choose not to use case sensitive file systems. In fact, I wonder if it's safe to test this tired to the OS at all...